lib/gears/wallpaper.lua

```diff
@@ -66,10 +66,12 @@ function wallpaper.prepare_context(s)
 
         -- Set the wallpaper (delayed)
         timer.delayed_call(function()
-            local paper = pending_wallpaper
-            pending_wallpaper = nil
-            wallpaper.set(paper.surface)
-            paper.surface:finish()
+            if pending_wallpaper then
+                local paper = pending_wallpaper
+                pending_wallpaper = nil
+                wallpaper.set(paper.surface)
+                paper.surface:finish()
```

Comment:
wont this need an "else" to start the delayed call again and a number of attempts before giving up? Right now I think this could cause the wallpaper to be lost.

Reply:
It doesn't on my system at least, I had thought about doing that but it seems like it's not necessary? I can add it to be extra defensive, but I believe it might cause the callback to be called infinitely.

Reply:
sorry that was stupid. I'm testing it more thoroughly now